src/graphics/window/event.rs

```diff
@@ -46,15 +46,50 @@ impl EventLoop {
                         state,
                         button,
                     })),
+                    winit::WindowEvent::MouseWheel { delta, .. } => match delta
+                    {
+                        winit::MouseScrollDelta::LineDelta(x, y) => {
+                            f(Event::Input(input::Event::MouseWheel {
+                                delta_x: x,
+                                delta_y: y,
+                            }))
+                        }
+                        _ => {}
+                    },
+
+                    winit::WindowEvent::ReceivedCharacter(codepoint) => {
+                        f(Event::Input(input::Event::TextInput {
+                            character: codepoint,
+                        }))
+                    }
                     winit::WindowEvent::CursorMoved { position, .. } => {
                         f(Event::CursorMoved(position))
                     }
-                    winit::WindowEvent::CloseRequested => {
+                    winit::WindowEvent::CursorEntered { .. } => {
+                        f(Event::Input(input::Event::CursorEntered))
+                    }
+                    winit::WindowEvent::CursorLeft { .. } => {
+                        f(Event::Input(input::Event::CursorLeft))
+                    }
+                    winit::WindowEvent::CloseRequested { .. } => {
                         f(Event::CloseRequested)
                     }
                     winit::WindowEvent::Resized(logical_size) => {
                         f(Event::Resized(logical_size))
                     }
+                    winit::WindowEvent::Focused(focus) => {
+                        f(Event::Input(if focus == true {
+                            input::Event::WindowFocused
+                        } else {
+                            input::Event::WindowUnfocused
+                        }))
+                    }
+                    winit::WindowEvent::Moved(
+                        winit::dpi::LogicalPosition { x, y },
+                    ) => f(Event::Input(input::Event::WindowMoved {
+                        x: x as f32,
+                        y: y as f32,
+                    })),
```

Comment:
Right now, Coffee exposes all the positions in physical coordinates. Although this may change soon (check out #6), we should convert this into physical coordinates to stay consistent.

The `CursorMoved` event does this and may serve us as a guide. We simply need to add a new `window::Event` (we can name it `Moved`) and then convert it into an `input::Event` using `window.dpi()` in the game loop, like we do right [here](https://github.com/hecrj/coffee/blob/master/src/lib.rs#L294) for `CursorMoved`.

I am aware this is a bit awkward, but if we end up doing #6 this complexity will go away!